tests/installation/ipxe_install.pm

```diff
@@ -0,0 +1,112 @@
+# SUSE's openQA tests
+#
+# Copyright © 2012-2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify installation starts and is in progress
+# Maintainer: Michael Moese <mmoese@suse.de>
+
+use strict;
+use warnings;
+
+use testapi;
+use bmwqemu;
+use base "y2logsstep";
+
+use HTTP::Tiny;
+use IPC::Run;
+use Socket;
+use Time::HiRes qw(sleep);
+
+sub ipmitool {
+    my ($cmd) = @_;
+
+    my @cmd = ('ipmitool', '-I', 'lanplus', '-H', $bmwqemu::vars{IPMI_HOSTNAME}, '-U', $bmwqemu::vars{IPMI_USER}, '-P', $bmwqemu::vars{IPMI_PASSWORD});
+    push(@cmd, split(/ /, $cmd));
+
+    my ($stdin, $stdout, $stderr, $ret);
+    print @cmd;
+    $ret = IPC::Run::run(\@cmd, \$stdin, \$stdout, \$stderr);
+    chomp $stdout;
+    chomp $stderr;
+
+    die join(' ', @cmd) . ": $stderr" unless ($ret);
+    bmwqemu::diag("IPMI: $stdout");
+    return $stdout;
+}
+
+sub poweroff_host {
+    ipmitool("chassis power off");
+    while (1) {
```

Comment:
What about adding some timeout to this loop? It shouldn't be the case, but if the machine doesn't respond to the ipmi command, this will hang forever.

Reply:
Well, not forever. At some point, the entire test will time out. And in case the BMC does not respond to commands, we don't block the machine - no other test will be able to be executed there as well.

Reply:
right, but what if we have a job with MAX_JOB_TIME 10 hours for example and this fails powering on/off? It would be a pity waiting 10 hours for the job timeout.
It doesn't have to be done now, but it's just a suggestion for later improvement :)